<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1270</width>
    <height>413</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QColumnView" name="columnView_16">
    <property name="geometry">
     <rect>
      <x>940</x>
      <y>110</y>
      <width>51</width>
      <height>91</height>
     </rect>
    </property>
   </widget>
   <widget class="QColumnView" name="columnView_10">
    <property name="geometry">
     <rect>
      <x>580</x>
      <y>110</y>
      <width>51</width>
      <height>91</height>
     </rect>
    </property>
   </widget>
   <widget class="QColumnView" name="columnView_6">
    <property name="geometry">
     <rect>
      <x>340</x>
      <y>110</y>
      <width>51</width>
      <height>91</height>
     </rect>
    </property>
   </widget>
   <widget class="QColumnView" name="columnView_13">
    <property name="geometry">
     <rect>
      <x>760</x>
      <y>110</y>
      <width>51</width>
      <height>91</height>
     </rect>
    </property>
   </widget>
   <widget class="QColumnView" name="columnView_19">
    <property name="geometry">
     <rect>
      <x>1120</x>
      <y>110</y>
      <width>51</width>
      <height>91</height>
     </rect>
    </property>
   </widget>
   <widget class="QColumnView" name="columnView_14">
    <property name="geometry">
     <rect>
      <x>820</x>
      <y>110</y>
      <width>51</width>
      <height>91</height>
     </rect>
    </property>
   </widget>
   <widget class="QColumnView" name="columnView_20">
    <property name="geometry">
     <rect>
      <x>1180</x>
      <y>110</y>
      <width>51</width>
      <height>91</height>
     </rect>
    </property>
   </widget>
   <widget class="QColumnView" name="columnView_8">
    <property name="geometry">
     <rect>
      <x>460</x>
      <y>110</y>
      <width>51</width>
      <height>91</height>
     </rect>
    </property>
   </widget>
   <widget class="QColumnView" name="columnView">
    <property name="geometry">
     <rect>
      <x>40</x>
      <y>110</y>
      <width>51</width>
      <height>91</height>
     </rect>
    </property>
    <property name="frameShape">
     <enum>QFrame::StyledPanel</enum>
    </property>
   </widget>
   <widget class="QColumnView" name="columnView_11">
    <property name="geometry">
     <rect>
      <x>640</x>
      <y>110</y>
      <width>51</width>
      <height>91</height>
     </rect>
    </property>
   </widget>
   <widget class="QColumnView" name="columnView_3">
    <property name="geometry">
     <rect>
      <x>160</x>
      <y>110</y>
      <width>51</width>
      <height>91</height>
     </rect>
    </property>
   </widget>
   <widget class="QColumnView" name="columnView_7">
    <property name="geometry">
     <rect>
      <x>400</x>
      <y>110</y>
      <width>51</width>
      <height>91</height>
     </rect>
    </property>
   </widget>
   <widget class="QColumnView" name="columnView_12">
    <property name="geometry">
     <rect>
      <x>700</x>
      <y>110</y>
      <width>51</width>
      <height>91</height>
     </rect>
    </property>
   </widget>
   <widget class="QColumnView" name="columnView_18">
    <property name="geometry">
     <rect>
      <x>1060</x>
      <y>110</y>
      <width>51</width>
      <height>91</height>
     </rect>
    </property>
   </widget>
   <widget class="QColumnView" name="columnView_15">
    <property name="geometry">
     <rect>
      <x>880</x>
      <y>110</y>
      <width>51</width>
      <height>91</height>
     </rect>
    </property>
   </widget>
   <widget class="QColumnView" name="columnView_9">
    <property name="geometry">
     <rect>
      <x>520</x>
      <y>110</y>
      <width>51</width>
      <height>91</height>
     </rect>
    </property>
   </widget>
   <widget class="QColumnView" name="columnView_2">
    <property name="geometry">
     <rect>
      <x>100</x>
      <y>110</y>
      <width>51</width>
      <height>91</height>
     </rect>
    </property>
   </widget>
   <widget class="QColumnView" name="columnView_17">
    <property name="geometry">
     <rect>
      <x>1000</x>
      <y>110</y>
      <width>51</width>
      <height>91</height>
     </rect>
    </property>
   </widget>
   <widget class="QColumnView" name="columnView_4">
    <property name="geometry">
     <rect>
      <x>220</x>
      <y>110</y>
      <width>51</width>
      <height>91</height>
     </rect>
    </property>
   </widget>
   <widget class="QColumnView" name="columnView_5">
    <property name="geometry">
     <rect>
      <x>280</x>
      <y>110</y>
      <width>51</width>
      <height>91</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="label">
    <property name="geometry">
     <rect>
      <x>60</x>
      <y>140</y>
      <width>21</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>20</pointsize>
     </font>
    </property>
    <property name="text">
     <string>1</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_2">
    <property name="geometry">
     <rect>
      <x>120</x>
      <y>140</y>
      <width>21</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>20</pointsize>
     </font>
    </property>
    <property name="text">
     <string>2</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_3">
    <property name="geometry">
     <rect>
      <x>180</x>
      <y>140</y>
      <width>21</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>20</pointsize>
     </font>
    </property>
    <property name="text">
     <string>3</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_4">
    <property name="geometry">
     <rect>
      <x>240</x>
      <y>140</y>
      <width>21</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>20</pointsize>
     </font>
    </property>
    <property name="text">
     <string>4</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_5">
    <property name="geometry">
     <rect>
      <x>300</x>
      <y>140</y>
      <width>21</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>20</pointsize>
     </font>
    </property>
    <property name="text">
     <string>5</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_6">
    <property name="geometry">
     <rect>
      <x>360</x>
      <y>140</y>
      <width>21</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>20</pointsize>
     </font>
    </property>
    <property name="text">
     <string>6</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_7">
    <property name="geometry">
     <rect>
      <x>420</x>
      <y>140</y>
      <width>21</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>20</pointsize>
     </font>
    </property>
    <property name="text">
     <string>7</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_8">
    <property name="geometry">
     <rect>
      <x>480</x>
      <y>140</y>
      <width>21</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>20</pointsize>
     </font>
    </property>
    <property name="text">
     <string>8</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_9">
    <property name="geometry">
     <rect>
      <x>540</x>
      <y>140</y>
      <width>21</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>20</pointsize>
     </font>
    </property>
    <property name="text">
     <string>9</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_10">
    <property name="geometry">
     <rect>
      <x>590</x>
      <y>140</y>
      <width>41</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>20</pointsize>
     </font>
    </property>
    <property name="text">
     <string>10</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_11">
    <property name="geometry">
     <rect>
      <x>650</x>
      <y>140</y>
      <width>31</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>20</pointsize>
     </font>
    </property>
    <property name="text">
     <string>11</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_12">
    <property name="geometry">
     <rect>
      <x>710</x>
      <y>140</y>
      <width>41</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>20</pointsize>
     </font>
    </property>
    <property name="text">
     <string>12</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_13">
    <property name="geometry">
     <rect>
      <x>770</x>
      <y>140</y>
      <width>41</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>20</pointsize>
     </font>
    </property>
    <property name="text">
     <string>13</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_14">
    <property name="geometry">
     <rect>
      <x>830</x>
      <y>140</y>
      <width>41</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>20</pointsize>
     </font>
    </property>
    <property name="text">
     <string>14</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_15">
    <property name="geometry">
     <rect>
      <x>890</x>
      <y>140</y>
      <width>41</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>20</pointsize>
     </font>
    </property>
    <property name="text">
     <string>15</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_16">
    <property name="geometry">
     <rect>
      <x>950</x>
      <y>140</y>
      <width>41</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>20</pointsize>
     </font>
    </property>
    <property name="text">
     <string>16</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_17">
    <property name="geometry">
     <rect>
      <x>1010</x>
      <y>140</y>
      <width>41</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>20</pointsize>
     </font>
    </property>
    <property name="text">
     <string>17</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_18">
    <property name="geometry">
     <rect>
      <x>1070</x>
      <y>140</y>
      <width>41</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>20</pointsize>
     </font>
    </property>
    <property name="text">
     <string>18</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_19">
    <property name="geometry">
     <rect>
      <x>1130</x>
      <y>140</y>
      <width>41</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>20</pointsize>
     </font>
    </property>
    <property name="text">
     <string>19</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_20">
    <property name="geometry">
     <rect>
      <x>1190</x>
      <y>140</y>
      <width>41</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>20</pointsize>
     </font>
    </property>
    <property name="text">
     <string>20</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_21">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>10</y>
      <width>1081</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>15</pointsize>
     </font>
    </property>
    <property name="text">
     <string>生产者生产指针(生产者编号为0,1,2，…，上部的数字代表生产者编号，所在位置即为指针位置)</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_22">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>270</y>
      <width>1071</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>15</pointsize>
     </font>
    </property>
    <property name="text">
     <string>消费者消费指针(消费者编号为0,1,2，…，下部的数字代表消费者编号，所在位置即为指针位置)</string>
    </property>
   </widget>
   <widget class="QTextEdit" name="textEdit">
    <property name="geometry">
     <rect>
      <x>40</x>
      <y>50</y>
      <width>51</width>
      <height>41</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>7</pointsize>
     </font>
    </property>
    <property name="html">
     <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'SimSun'; font-size:7pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot;-qt-paragraph-type:empty; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px; font-family:'SimSun'; font-size:8pt;&quot;&gt;&lt;br /&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
    </property>
   </widget>
   <widget class="QTextEdit" name="textEdit_2">
    <property name="geometry">
     <rect>
      <x>100</x>
      <y>50</y>
      <width>51</width>
      <height>41</height>
     </rect>
    </property>
   </widget>
   <widget class="QTextEdit" name="textEdit_3">
    <property name="geometry">
     <rect>
      <x>160</x>
      <y>50</y>
      <width>51</width>
      <height>41</height>
     </rect>
    </property>
   </widget>
   <widget class="QTextEdit" name="textEdit_4">
    <property name="geometry">
     <rect>
      <x>220</x>
      <y>50</y>
      <width>51</width>
      <height>41</height>
     </rect>
    </property>
   </widget>
   <widget class="QTextEdit" name="textEdit_5">
    <property name="geometry">
     <rect>
      <x>280</x>
      <y>50</y>
      <width>51</width>
      <height>41</height>
     </rect>
    </property>
   </widget>
   <widget class="QTextEdit" name="textEdit_6">
    <property name="geometry">
     <rect>
      <x>340</x>
      <y>50</y>
      <width>51</width>
      <height>41</height>
     </rect>
    </property>
   </widget>
   <widget class="QTextEdit" name="textEdit_7">
    <property name="geometry">
     <rect>
      <x>400</x>
      <y>50</y>
      <width>51</width>
      <height>41</height>
     </rect>
    </property>
   </widget>
   <widget class="QTextEdit" name="textEdit_8">
    <property name="geometry">
     <rect>
      <x>460</x>
      <y>50</y>
      <width>51</width>
      <height>41</height>
     </rect>
    </property>
   </widget>
   <widget class="QTextEdit" name="textEdit_9">
    <property name="geometry">
     <rect>
      <x>520</x>
      <y>50</y>
      <width>51</width>
      <height>41</height>
     </rect>
    </property>
   </widget>
   <widget class="QTextEdit" name="textEdit_10">
    <property name="geometry">
     <rect>
      <x>580</x>
      <y>50</y>
      <width>51</width>
      <height>41</height>
     </rect>
    </property>
   </widget>
   <widget class="QTextEdit" name="textEdit_11">
    <property name="geometry">
     <rect>
      <x>640</x>
      <y>50</y>
      <width>51</width>
      <height>41</height>
     </rect>
    </property>
   </widget>
   <widget class="QTextEdit" name="textEdit_12">
    <property name="geometry">
     <rect>
      <x>700</x>
      <y>50</y>
      <width>51</width>
      <height>41</height>
     </rect>
    </property>
   </widget>
   <widget class="QTextEdit" name="textEdit_13">
    <property name="geometry">
     <rect>
      <x>760</x>
      <y>50</y>
      <width>51</width>
      <height>41</height>
     </rect>
    </property>
   </widget>
   <widget class="QTextEdit" name="textEdit_14">
    <property name="geometry">
     <rect>
      <x>820</x>
      <y>50</y>
      <width>51</width>
      <height>41</height>
     </rect>
    </property>
   </widget>
   <widget class="QTextEdit" name="textEdit_15">
    <property name="geometry">
     <rect>
      <x>880</x>
      <y>50</y>
      <width>51</width>
      <height>41</height>
     </rect>
    </property>
   </widget>
   <widget class="QTextEdit" name="textEdit_16">
    <property name="geometry">
     <rect>
      <x>940</x>
      <y>50</y>
      <width>51</width>
      <height>41</height>
     </rect>
    </property>
   </widget>
   <widget class="QTextEdit" name="textEdit_17">
    <property name="geometry">
     <rect>
      <x>1000</x>
      <y>50</y>
      <width>51</width>
      <height>41</height>
     </rect>
    </property>
   </widget>
   <widget class="QTextEdit" name="textEdit_18">
    <property name="geometry">
     <rect>
      <x>1060</x>
      <y>50</y>
      <width>51</width>
      <height>41</height>
     </rect>
    </property>
   </widget>
   <widget class="QTextEdit" name="textEdit_19">
    <property name="geometry">
     <rect>
      <x>1120</x>
      <y>50</y>
      <width>51</width>
      <height>41</height>
     </rect>
    </property>
   </widget>
   <widget class="QTextEdit" name="textEdit_20">
    <property name="geometry">
     <rect>
      <x>1180</x>
      <y>50</y>
      <width>51</width>
      <height>41</height>
     </rect>
    </property>
   </widget>
   <widget class="QTextEdit" name="textEdit_21">
    <property name="geometry">
     <rect>
      <x>40</x>
      <y>220</y>
      <width>51</width>
      <height>41</height>
     </rect>
    </property>
   </widget>
   <widget class="QTextEdit" name="textEdit_22">
    <property name="geometry">
     <rect>
      <x>100</x>
      <y>220</y>
      <width>51</width>
      <height>41</height>
     </rect>
    </property>
   </widget>
   <widget class="QTextEdit" name="textEdit_23">
    <property name="geometry">
     <rect>
      <x>160</x>
      <y>220</y>
      <width>51</width>
      <height>41</height>
     </rect>
    </property>
   </widget>
   <widget class="QTextEdit" name="textEdit_24">
    <property name="geometry">
     <rect>
      <x>220</x>
      <y>220</y>
      <width>51</width>
      <height>41</height>
     </rect>
    </property>
   </widget>
   <widget class="QTextEdit" name="textEdit_25">
    <property name="geometry">
     <rect>
      <x>280</x>
      <y>220</y>
      <width>51</width>
      <height>41</height>
     </rect>
    </property>
   </widget>
   <widget class="QTextEdit" name="textEdit_26">
    <property name="geometry">
     <rect>
      <x>340</x>
      <y>220</y>
      <width>51</width>
      <height>41</height>
     </rect>
    </property>
   </widget>
   <widget class="QTextEdit" name="textEdit_27">
    <property name="geometry">
     <rect>
      <x>400</x>
      <y>220</y>
      <width>51</width>
      <height>41</height>
     </rect>
    </property>
   </widget>
   <widget class="QTextEdit" name="textEdit_28">
    <property name="geometry">
     <rect>
      <x>460</x>
      <y>220</y>
      <width>51</width>
      <height>41</height>
     </rect>
    </property>
   </widget>
   <widget class="QTextEdit" name="textEdit_29">
    <property name="geometry">
     <rect>
      <x>520</x>
      <y>220</y>
      <width>51</width>
      <height>41</height>
     </rect>
    </property>
   </widget>
   <widget class="QTextEdit" name="textEdit_30">
    <property name="geometry">
     <rect>
      <x>580</x>
      <y>220</y>
      <width>51</width>
      <height>41</height>
     </rect>
    </property>
   </widget>
   <widget class="QTextEdit" name="textEdit_31">
    <property name="geometry">
     <rect>
      <x>640</x>
      <y>220</y>
      <width>51</width>
      <height>41</height>
     </rect>
    </property>
   </widget>
   <widget class="QTextEdit" name="textEdit_32">
    <property name="geometry">
     <rect>
      <x>700</x>
      <y>220</y>
      <width>51</width>
      <height>41</height>
     </rect>
    </property>
   </widget>
   <widget class="QTextEdit" name="textEdit_33">
    <property name="geometry">
     <rect>
      <x>760</x>
      <y>220</y>
      <width>51</width>
      <height>41</height>
     </rect>
    </property>
   </widget>
   <widget class="QTextEdit" name="textEdit_34">
    <property name="geometry">
     <rect>
      <x>820</x>
      <y>220</y>
      <width>51</width>
      <height>41</height>
     </rect>
    </property>
   </widget>
   <widget class="QTextEdit" name="textEdit_35">
    <property name="geometry">
     <rect>
      <x>880</x>
      <y>220</y>
      <width>51</width>
      <height>41</height>
     </rect>
    </property>
   </widget>
   <widget class="QTextEdit" name="textEdit_36">
    <property name="geometry">
     <rect>
      <x>940</x>
      <y>220</y>
      <width>51</width>
      <height>41</height>
     </rect>
    </property>
   </widget>
   <widget class="QTextEdit" name="textEdit_37">
    <property name="geometry">
     <rect>
      <x>1000</x>
      <y>220</y>
      <width>51</width>
      <height>41</height>
     </rect>
    </property>
   </widget>
   <widget class="QTextEdit" name="textEdit_38">
    <property name="geometry">
     <rect>
      <x>1060</x>
      <y>220</y>
      <width>51</width>
      <height>41</height>
     </rect>
    </property>
   </widget>
   <widget class="QTextEdit" name="textEdit_39">
    <property name="geometry">
     <rect>
      <x>1120</x>
      <y>220</y>
      <width>51</width>
      <height>41</height>
     </rect>
    </property>
   </widget>
   <widget class="QTextEdit" name="textEdit_40">
    <property name="geometry">
     <rect>
      <x>1180</x>
      <y>220</y>
      <width>51</width>
      <height>41</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="label_23">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>310</y>
      <width>591</width>
      <height>41</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>15</pointsize>
     </font>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1270</width>
     <height>26</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
